pkg/proxy/ipvs/proxier_test.go

```diff
@@ -652,12 +652,29 @@ func TestNodePortIPv4(t *testing.T) {
 				}},
 			},
 			expectedIptablesChains: netlinktest.ExpectedIptablesChain{
-				string(KubeNodePortChain): {{
-					JumpChain: string(KubeMarkMasqChain), MatchSet: kubeNodePortSetUDP,
-				}},
-				string(kubeServicesChain): {{
-					JumpChain: string(KubeNodePortChain), MatchSet: "",
-				}},
+				string(KubeNodePortChain): {
```

Comment:
why is this test affected if we only have changed the SCTP rules order?

Reply:
Previously the test only checked whether the target iptables rule exists or not and didn't care about the extra rules got from the system and their order, by using `checkIptables`. If we just add the two sctp related rules to `expectedIptablesChains`, it's not going to validate one rule is after the other and can be broken in the future without any test failure. So I add a new method `checkIptablesInOrder` which validates the `expectedIptablesChains` have the same number and order of rules as the ones got from the system. As a result, we need to add missing chains of this test accordingly.

Reply:
hmm, shouldn't that be the default behavior of ` checkIptables` ? 
that method is used in other places and order matters, i.e. we can't have a RETURN as first rule and mark the test as valid XD

Reply:
I was trying to touch less code. But making `checkIptables` check rule order for all tests makes sense to me. I have updated `checkIptables` and other tests accordingly.